lib/sgtn-client/api/translation.rb

```diff
@@ -86,22 +86,30 @@ def self.getTranslation(component, key, locale)
       end
 
       def self.get_cs(component, locale)
+        if locale.nil?
+          locale=SgtnClient::Config.configurations.default
+        end
+        locale = locale.to_s
         flocale = SgtnClient::LocaleUtil.fallback(locale)
         cache_key = SgtnClient::CacheUtil.get_cachekey(component, flocale)
         SgtnClient.logger.debug "[Translation][get_cs]cache_key=#{cache_key}"
         expired, items = SgtnClient::CacheUtil.get_cache(cache_key)
-        if items.nil? || expired
-          items = load(component, flocale)
-          if items.nil?
-            items = SgtnClient::Source.getSources(component, SgtnClient::Config.configurations.default)
-            SgtnClient::Core::Cache.put(cache_key, items, 60)
+        t = Thread.new {
```

Comment:
Isn't this part supposed to be in a different PR?

Reply:
the code logic isn't changed, i think it should be easily understood.

Reply:
Yes, but separate testing needs to happen for this. If this causes an issue, we don;t want the rest of the CloudHealth PR to be pulled due top this,